torch/csrc/distributed/c10d/Functional.cpp

```diff
@@ -216,6 +215,25 @@ at::Tensor all_to_all_single(
   return output;
 }
 
+at::Tensor& broadcast_(at::Tensor& input, int64_t src, std::string group_name) {
+  c10d::BroadcastOptions opts;
+  opts.rootRank = src;
+  std::vector<at::Tensor> inputs{input};
+
+  auto group = c10d::resolve_process_group(group_name);
+  auto work = group->broadcast(inputs, opts);
```

Comment:
do we need to check if the `input` is contiguous somewhere for the inplace version?

Reply:
Hey @wanchaol, this is performed in `ProcessGroupNCCL`:

https://github.com/pytorch/pytorch/blob/bc7f3efb0987ed02599a34bd688f04e10a8308a9/torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp#L2975